### PR TITLE
DRF test client request parameters

### LINKED ISSUE #813: Use Django rest framework client instead of django client when testing
## Problem
The test cases are using Django client which is much suited for emulating a browser, so the default request body is priority content type is multipart instead of json which is what we are using mostly on the frontend

## Proposed solution
DRF [test client](https://www.django-rest-framework.org/api-guide/testing/) should be used instead of django client in order to emulate an API call rather than a browser form, this avoids unnecessary json dumps when calling an API during testing

### PR DESCRIPTION
**Pull request by andruwwwka**
Fixes #813.

Use correct django-rest-framework test client parameters when sending json body payload

## Checklist
 
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the `master` branch of the repository
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by kgodey** [comment]
@silentninja Assigning this to you for review, please feel free to pull in others if needed.

**Comment by silentninja** [comment]
@andruwwwka Thanks for the PR. I think it would be better to specify `json` as the default request format and use `format` option for the requests that are multipart. @kgodey I would appreciate your comments on this

**Comment by kgodey** [comment]
@silentninja That sounds fine to me, it would reduce boilerplate code.

**Comment by andruwwwka** [comment]
@silentninja Hi! I update the PR.

File: mathesar/tests/api/test_record_api.py [hunk 1]
```diff
@@ -427,10 +429,11 @@ def test_record_list_sort_exceptions(create_table, client, exception):
 def test_record_list_group_exceptions(create_table, client, exception):
     table_name = f"NASA Record List {exception.__name__}"
     table = create_table(table_name)
-    group_by = json.dumps(["Center"])
+    data = {"group_count_by": ["Center"]}
```
Comment (silentninja) [inline mathesar/tests/api/test_record_api.py:432]: This should be a json string.

**Review by silentninja — CHANGES_REQUESTED** [review]
`get` data objects should be JSON string rather than the string representation from `f` string conversion as the backend expects a JSON object and `f` string outputs a urlencoded string which is not a valid JSON object. Moreover, test cases could fail if the object has nested objects as URL encoded objects won't have a clean conversion, so I think it is better to stick with json encoded string

**Comment by andruwwwka** [comment]
@silentninja I have corrected the code according to your comment

**Review by silentninja — APPROVED** [review]

**Comment by kgodey** [comment]
Thanks for your work on this, @andruwwwka!

**Pull request merged**